File: leetcode/AIcoding/1.Gemini-2.5/go_solutions/728-self-dividing-numbers.go
func selfDividingNumbers(left int, right int) []int {
    var result []int

    for i := left; i <= right; i++ {
        if isSelfDividing(i) {
            result = append(result, i)
        }
    }

    return result
}

func isSelfDividing(n int) bool {
    temp := n
    for temp > 0 {
        digit := temp % 10
        if digit == 0 || n % digit != 0 {
            return false
        }
        temp /= 10
    }
    return true
}

File: leetcode/AIcoding/1.Gemini-2.5/rust_solutions/728-self-dividing-numbers.rs
impl Solution {
    pub fn self_dividing_numbers(left: i32, right: i32) -> Vec<i32> {
        let mut result = Vec::new();
        for num in left..=right {
            if Self::is_self_dividing(num) {
                result.push(num);
            }
        }
        result
    }

    fn is_self_dividing(n: i32) -> bool {
        let mut temp = n;
        while temp > 0 {
            let digit = temp % 10;
            if digit == 0 || n % digit != 0 {
                return false;
            }
            temp /= 10;
        }
        true
    }
}